### DISCUSSION TITLE
ModelView in issues API

### DISCUSSION THREAD

**Discussion by VedankNaik** [iTwin/itwinjs-core#5159] [Q&A] [answered]
Hello, 
1. How get I get data required for [Model View](https://developer.bentley.com/apis/issues-v1/operations/create-issue/#model-view) object while creating issues? 
2. How can I get snap of viewer while creating issue and add as attachment? (as seen on imodelweb which attaches snap of viewer to issues and is displayed as thumbnail)

**Comment by pratikshan85** [comment, marked as answer]
Hi [VedankNaik](https://github.com/VedankNaik),

For the second query, Please go through the below link with the API call to add an attachment to an issue.You need to make 2 API (Add and Upload) call for adding attachment.
https://developer.bentley.com/apis/issues-v1/operations/add-attachment-to-issue/


To get the snap of viewer please refer to the below sample .
https://www.itwinjs.org/sandboxes/iTwinPlatform/Issues

Note : @ajmiam Can you please look at the first query and also feel free to add more information for the second question.

**Reply by VedankNaik** [reply]
Hello @pratikshan85,
I've checked adding attachments API, I'm able to add attachments. But what I am currently doing is create issue, manually capture the screen and upload image as attachment later. I wanted to know if there's any itwin feature that will allow me to capture the viewer and upload it while creating issue itself.

Thanks.

**Reply by pmconne** [reply]
@VedankNaik you can capture the contents of a viewport using [readImageBuffer](https://www.itwinjs.org/reference/core-frontend/views/viewport/readimagebuffer/).
Here's an extremely basic [debugging tool](https://github.com/iTwin/itwinjs-core/blob/master/test-apps/display-test-app/src/frontend/SaveImageTool.ts) illustrating how to do that.

**Reply by VedankNaik** [reply]
Thanks @pmconne. I'll try this out.
Anything for point 1?

**Reply by pmconne** [reply]
@jzbuchalski and/or @diegopinate ought to be able to help with how to specify the issue's modelView.
If you are using the saved views API you can specify the Id of a saved view.
Otherwise, we provided them some code a while back to produce the camera info, but I don't know where it ended up or what modifications were made to it.

**Reply by VedankNaik** [reply]
> @VedankNaik you can capture the contents of a viewport using [readImageBuffer](https://www.itwinjs.org/reference/core-frontend/views/viewport/readimagebuffer/). Here's an extremely basic [debugging tool](https://github.com/iTwin/itwinjs-core/blob/master/test-apps/display-test-app/src/frontend/SaveImageTool.ts) illustrating how to do that.

I was able to capture the contents of the viewport using the tool linked above. I also managed to create a ModelView object using functions available in the ViewState class.

Thanks.
